compiler/src/yul/mappers/expressions.rs

```diff
@@ -272,6 +272,27 @@ fn expr_attribute_self(
     }
 }
 
+fn expr_ternary(
+    context: &Context,
+    exp: &Spanned<fe::Expr>,
+) -> Result<yul::Expression, CompileError> {
+    if let fe::Expr::Ternary {
+        test,
+        if_expr,
+        else_expr,
+    } = &exp.node
+    {
+        let yul_test_expr = expr(context, test)?;
+        let yul_if_expr = expr_comp_operation(context, if_expr)?;
+        let yul_else_exp = expr(context, else_expr)?;
+
+        // TODO: Need to find a way to return an expression that comprises all 3 of
+        // above. Dummy return to make it compile.
+        return Ok(literal_expression! {(1)});
```

Comment:
>My understanding is that we are missing the appropriate macro in the yulstur crate, As I can't find a macro or helper function that generates a valid expression from those 3 expressions, Or maybe I am missing something.

I'm going to answer this here because there's a chance that I'll end up giving you a wrong or misleading answer and so I want to page in @g-r-a-n-t to give him the chance to correct me if I'm talking nonsense :sweat_smile: 

So, what I'm thinking is that this would be best mapped to a `switch` statement in YUL. You can see this being used here:

https://github.com/ethereum/fe/blob/c2681b844bf441ee296356ea6ab41f86fbd1e125/compiler/src/yul/mappers/functions.rs#L107-L111

Now, the difference here would be the `switch` is a statement so it doesn't resolve to any value. So, what I'm thinking is that we would need to map this to a bigger code out to something like this:

Pseudo-YUL
```
switch test
case 0 { $result := <if-expression> }
case 1 { $result := <else-expression }
result
```

So, basically we would generate a switch *statement* that introduces a variable to store the result of each possible arm followed by an *expression* that simply resolves to whatever is behind `$result`.

Without getting my hand on the code to actually try it, this is what I think I would try first.

@g-r-a-n-t  Does this make sense?

Reply:
> So, basically we would generate a switch statement that introduces a variable to store the result of each possible arm followed by an expression that simply resolves to whatever is behind $result.

The problem I saw is that we `switch!` is a statement I can't use to generate the expression so do you suggest using `yul::Switch` to generate the expression as you suggest above directly in the mappers instead of using the macros. 

Why do we need to store the result isn't we recursively generate the cases for it as per the given expressions.

Reply:
>The problem I saw is that we switch! is a statement I can't use to generate the expression

Correct. That's why I was thinking we could generate a switch that stores the result and then have that followed by an *expression* that just evaluates to whatever is behind that identifier. But as I'm looking over the various options, it's not immediately obvious to me how that could be done. @g-r-a-n-t can you help us out?

Reply:
Ah, yeah this is tricky. In Yul we can't wrap statements within an expression like you can in Rust... so we fundamentally can't map this entirely on its own.

```rust
// an expression in Rust
{ 
  let a = 42;
  a
}
```

The work around for this should be to create a helper function that contains the terany statements and takes the relevant expressions as parameters. I'm thinking it should maybe look something like this:

```javascript
function teranary(test, if_expression, else_expression) -> result {
  switch test
  case 0 { result := <if-expression> }
  case 1 { result := <else-expression }
}
```

This ternary function would be included within the [standard](https://github.com/ethereum/fe/blob/c2681b844bf441ee296356ea6ab41f86fbd1e125/compiler/src/yul/runtime/functions.rs#L5-L20) set of functions made available during runtime, then when we map a ternary expression, it should looks something like this:

`1 if input > 5 else 0` -> `ternary(input > 5, 1, 0)` 

Hopefully this works and thank you for the commits. This was supposed to be an easy one :laughing:

Reply:
I actually wouldn't really call it a work around. Languages that map to Yul are supposed to be human-readable, so it's good to keep expressions short.

Reply:
Thanks, @g-r-a-n-t  & @cburgdorf it works.